packages/pubsub/src/subscription.js

```diff
@@ -592,30 +350,154 @@ Subscription.prototype.decorateMessage_ = function(message) {
  *   var apiResponse = data[0];
  * });
  */
-Subscription.prototype.delete = function(callback) {
+Subscription.prototype.delete = function(gaxOpts, callback) {
   var self = this;
 
-  callback = callback || common.util.noop;
+  if (is.fn(gaxOpts)) {
+    callback = gaxOpts;
+    gaxOpts = {};
+  }
 
-  var protoOpts = {
-    service: 'Subscriber',
-    method: 'deleteSubscription'
+  var reqOpts = {
+    subscription: this.name
   };
 
+  this.request({
+    client: 'subscriberClient',
+    method: 'deleteSubscription',
+    reqOpts: reqOpts,
+    gaxOpts: gaxOpts
+  }, function(err, resp) {
+    if (!err) {
+      self.closeConnection_();
```

Comment:
> I'm a little lost on why we would want to call openConnection when it's already open.

I'm not suggesting that you should call openConnection. I'm merely describing the code that is written. It is good that openConnection is not being called.

> But this should throw an error if the subscription is deleted, right?

Why is it going to throw an error? The code will make an RPC to Pub/Sub. Then the code will asynchronously receive an error response from the service sometime later.

> We would then have 1000 message listeners hanging on to a closed connection.

I'm pointing out that having non-zero message listeners on a Subscription object with a closed connection is a possible state. But it's not a state that makes sense.

> I'm confused by this. Yes, we open on the 1st, because the user is looking for messages and a connection does not exist. Why would we open additional connections for all other event listeners?

I absolutely agree with you. We should call openConnection on the 1st message listener, and not on any of the subsequent ones. So why then are we calling openConnection on the 1001st message listener?

So my main question to you is: Why is it possible to get into a state where there are non-zero message listener but the connection is closed? That is what I don't understand.

It seems like the transition from 0->1 message listeners should trigger an openConnection and the transition from 1->0 message listeners should trigger a closeConnection... and nothing else. Then we preserve a straightforward invariant of "there's an open connection if and only if there are message listeners."